### PR TITLE
Added experimental option to one-page build that doesn't use XHR to load assets

### PR DESCRIPTION
**Pull request by yaustar**
Looks like Facebook Playable ads don't allow XMLHttpRequests
image

This option adds an engine patch to http.get to not use XMLHttpRequests and handles decoding the base64 data in the get function.

### CODE REVIEW

File: one-page.js [hunk 1]
```diff
@@ -80,6 +101,10 @@ function inlineAssets(projectPath) {
                     var urlSplit = url.split('.');
                     var extension = urlSplit[urlSplit.length - 1];
 
+                    if (asset.name == 'cocktail.json') {
+                        console.log('foo');
+                    }
+
```
Comment (willeastcott) [inline one-page.js:107]: Did you mean to include these lines?

Comment (yaustar) [reply one-page.js]: I did not 😆 . Removed

**Review by willeastcott — APPROVED** [review]

**Comment by yaustar** [comment]
@willeastcott I've made a few changes to cater for html and css files. Can you give a quick re-review please?

File: one-page.js [hunk 2]
```diff
@@ -112,11 +133,17 @@ function inlineAssets(projectPath) {
...
+                            // The model and animation loader assumes that the base64 URL will be loaded as a binary
+                            if ((asset.type !== 'model' && asset.type !== 'animation')) {
                                 mimeprefix = "data:application/json";
                             }
                         } break;
 
+                        case "css":
+                        case "html": {
+                            mimeprefix = "data:text/plain";
+                        } break;
```
Comment (willeastcott) [inline one-page.js:145]: The curly braces around all your case statements are unnecessary.

**Review by willeastcott — APPROVED** [review]
Re-approved with one comment.

File: one-page.js (hunk 2 above)
Comment (yaustar) [reply one-page.js:145]: Ah, old habit of C++ code style to always use braces. Would you prefer them to be removed for consistency with other repos?

Comment (willeastcott) [reply one-page.js:145]: I would. But up to you. 😉

**Pull request merged**